blog/wdio-blog.md

```diff
@@ -0,0 +1,284 @@
+
+Let's configure our current WDIO test to run using SauceLabs simulators.
+
+**Step 1:** Install [WDIO Sauce Service](http://webdriver.io/guide/services/sauce.html)
+
+```console
+$ npm install wdio-sauce-service --save-dev
+```
+
+**Step 2:** Update wdio.conf file.
+
+```diff
+- services: ['appium'],
++ services: ['sauce'],
+- host: 'ondemand.saucelabs.com',
+- port: 4723,
++ port: 80,
++ user: SAUCE_USERNAME,
++ key: SAUCE_ACCESS_KEY
+```
+
+Use [Platform configurator by SauceLabs](https://wiki.saucelabs.com/display/DOCS/Platform+Configurator) to update your Capabilities if needed. 
+
+
+**Step 3:** Upload `.app` or `.apk` file to SauceStorage or any other accessible endpoint. Follow docs here: [Uploading Mobile Applications to Sauce Storage for Testing](https://wiki.saucelabs.com/display/DOCS/Uploading+Mobile+Applications+to+Sauce+Storage+for+Testing)
+
+**Update App path in capabilities:** 
+```diff
+- app: <app path>
++ app: sauce-storage:myapp.zip or app link
+```
+
+**Run the test:**
+```console
+$ wdio wdio.conf.js
+```
+
+Check [SauceLabs Dashboard](https://saucelabs.com/beta/dashboard/tests) to make sure test has run. WDIO Sauce service automatically sets Test labels and results. 
+
+
+## Part 3: Running on Real Devices.
+SauceLab's also provides Real device testing solution through TestObject platform. With some minor changes to above wdio.conf, we can run UI tests on Real devices.
+
+Firstly, create a project in TestObject dashboard and upload our .ipa or .apk file. 
+
+There is no official WebDriverIO TestObject service, and at the time of writing this blog, SauceLabs and TestObject do not share same API calls. 
+
+**Update `wdio.conf` file:**
+```diff
+- services: ['sauce'],
++ protocol: 'https',
++ host: 'us1.appium.testobject.com',
+- port: '80',
++ port: '443',
++ path: '/api/appium/wd/hub',
+capabilities: {
+  ...
+-  app: 'sauce-storage:myapp.zip'
++  testobject_api_key: <TESTOBJECT_ACCESS_KEY>,
++  testobject_app_id: <APP_ID>,
+}
+```
+
+Refer [Appium Capabilities for Real Device Testing](https://wiki.saucelabs.com/display/DOCS/Appium+Capabilities+for+Real+Device+Testing) to update your Capabilities if needed. 
+
+**Run the UI test:**
+
+```console
+$ wdio wdio.conf
+```
+
+Check [TestObject Dashboard](https://app.testobject.com/) to make sure test has run. WDIO does not update Test labels and results for TestObject. 
+Here are some references that can let you update the test results: https://github.com/pizzasaurusrex/TestObject or https://gist.github.com/RTsGIT/0e8c0f89671a8a563a7463f8c1ff0413
+
+
+Hope the above discussion helps you set up some infrastructure for running WDIO UI tests locally and remotely for Emulators/Real devices. 
+
+## Part 4: Timeouts
+While we started to write UI tests and run them as part of CICD, Timeouts played a very important role in making sure UI tests are stable. There are various timeouts in play here at each level of tech stack: WDIO, Mocha/Jasmine, Appium and SauceLabs/TestObject
```

Comment:
"While we..." -> "When we..."

lowercase "timeouts"

"tests were stable" (make tense consistent with the rest of the sentence)

Period at the very end